webfrontend/src/components/popup/request-tag-modal.tsx

```diff
@@ -0,0 +1,227 @@
+            setIsLoading(false);
+            if (response.status !== 200) {
+              triggerErrorMessage();
+            } else {
+              setTag("");
+              setDescription("");
+              setAttach(false);
+              setImagePreview("");
+              const text = await response.text();
+              if (text === "tag already present/requested") {
+                setTagRequestMessageContent("Tag is already present/requested");
+              } else {
+                setTagRequestMessageContent(
+                  "Your request is sent, our admin will review your request."
+                );
+              }
+              setShowMessage(true);
+            }
+          }
+        );
+      }
+    },
+    [attach, description, imageToBeSubmitted, isLoading, tag]
+  );
+
+  const handleCloseMessage = useCallback(() => {
+    setShowMessage(false);
+    setTagRequestMessageContent("");
+    if (
+      tagRequestMessageContent ===
+      "Your request is sent, our admin will review your request."
+    ) {
+      handleClose();
+    }
+  }, [handleClose, tagRequestMessageContent, setTagRequestMessageContent]);
+
+  return (
+    <>
+      <img src={imgLink} alt={"request-tag"} onClick={handleShow} />
+
+      <Modal className="request-tag-modal" show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Request Tag</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form onSubmit={(e) => submitForm(e)} className="request-tag-form">
+            <div>
+              <label className="form-label">
+                Tag name <strong className="asterisk">*</strong>
+              </label>
+              <input
+                ref={tagInput}
+                type="text"
+                className="form-control"
+                value={tag}
+                onChange={(event) => {
+                  setTag(event.target.value.slice(0, 25));
+                }}
+              />
+            </div>
+
+            <div id="tag-request-icon-preview" className="mt-3">
+              <div>Icon:</div>
+              {imagePreview ? (
+                <img
+                  src={imagePreview}
+                  alt="request-tag-icon"
+                  style={{ height: "25px" }}
+                />
+              ) : (
+                ""
+              )}
+            </div>
+            <div className="mt-3">
+              <button
+                className="add-image-label"
+                onClick={(e) => {
+                  e.preventDefault();
+                  imageInput.current!.click();
+                }}
+              >
+                <img src="/static/media/add-image-icon.png" alt="add-icon" />
+              </button>
+              <input
+                ref={imageInput}
+                type="file"
+                className="form-control img-input"
+                accept="image/*"
+                onChange={(e) => handleAddImage(e)}
```

Comment:
`onChange={handleAddImage}` instead. This is so that whenever `handleAddImage` definition is updated, this callback is updated. Note that callbacks in React is only defined for the first render if it is not wrapped inside a `useCallback` hook.

Reply:
Please make corrections on other callbacks in this file as well, not just this highlighted one.